tests/integration/lpt/test_lpt.py

```diff
@@ -233,3 +231,27 @@ def test_aggregate_holdings(self):
             lambda left: self.fail(lpt.display_error(left)),
             lambda right: self.assertIsInstance(right, DataFrame),
         )
+
+    def test_aggregate_group_holdings(self):
+
+        if not self.target_portfolios_exist():
+            self.skipTest("missing target portfolios")
+
+        qah.process_args(
+            self.api,
+            qah.parse(
+                args=[
+                    "Finbourne-Examples",
+                    "US-FI",
+                    "2020-01-01",
+                    "--group",
+                    "--pricing-scope",
+                    "Finbourne-Examples",
+                    "--recipe",
+                    "FinbourneExamplesRecipeMidThenBid",
+                ]
+            ),
+        ).match(
+            lambda left: self.fail(lpt.display_error(left)),
+            lambda right: self.assertIsInstance(right, DataFrame),
```

Comment:
Am I right in saying this request could fail and the test would still pass?

Reply:
no, `process_arg` returns an `Either`, if the function fails it returns a `left` value and if successful it returns `right`. `match` then matches on these values, so if the call fails it will match on `left` and `self.fail`